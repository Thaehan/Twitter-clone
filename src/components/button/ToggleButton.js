import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';
import {
  BLACK_TEXT_COLOR,
  BACKGROUND_COLOR,
} from '../../styles/Style';
export default function ToggleButton({
  style,
  onPress,
  trueText,
  falseText,
  isTrue = false
}) {
  return (
    <View style={style}>
      <TouchableOpacity
        style={[styles.button,
        {
          backgroundColor: isTrue ? BACKGROUND_COLOR : BLACK_TEXT_COLOR
        }]}
        onPress={onPress}
      >
        <Text style={[styles.buttonText, {
          color: (!isTrue) ? BACKGROUND_COLOR : BLACK_TEXT_COLOR

        }]}>{isTrue ? trueText : falseText}</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  button: {
    alignSelf: 'center',
    borderColor: BLACK_TEXT_COLOR,
    borderRadius: 100,
    borderWidth: 1,
  },
  buttonText: {
    fontWeight: 'bold',
    paddingBottom: 8,
    paddingLeft: 15,
    paddingRight: 15,
    paddingTop: 6,
  },
});
