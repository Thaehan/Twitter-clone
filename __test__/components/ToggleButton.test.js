import React from 'react';
// import renderer from 'react-test-renderer';\
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import store from '../../src/redux/store';
import ToggleButton from '../../src/components/button/ToggleButton';

jest.useFakeTimers();
jest.mock(
  'react-native/Libraries/Animated/NativeAnimatedHelper'
);

describe('<ToggleButton />', () => {
  it('Render ToggleButton', () => {
    const testComponent = render(<ToggleButton />);
    expect(testComponent.toJSON()).toMatchSnapshot();
    // testComponent.getByTestId('AvatarImage').tobe();
  });
});
