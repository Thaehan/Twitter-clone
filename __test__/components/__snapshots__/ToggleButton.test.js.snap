// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> Render ToggleButton 1`] = `
<View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#0F1419",
        "borderColor": "#0F1419",
        "borderRadius": 100,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontWeight": "bold",
            "paddingBottom": 8,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 6,
          },
          Object {
            "color": "white",
          },
        ]
      }
    />
  </View>
</View>
`;
